### ISSUE TITLE
bug: [react] ionViewDidEnter event fires only for one component in the view

### ISSUE THREAD

**Issue by SatoAV** [ionic-team/ionic#19792]
Bug Report

**Ionic version:**

[x] **4.11.0**

**Current behavior:**
If I have several components wrapped in `withIonLifeCycle` HOC on one page, `ionViewDidEnter` event gets fired only on the first component, but not others.

**Expected behavior:**
If I undertstand the purpose of this event, all components in the view should fire `ionViewDidEnter` events.

**Steps to reproduce:**
Clone the test repo:
https://github.com/SatoAV/ionic-test

Run `npm i`, then `npm run start`.
Open Test Page from the menu. If you open the console, you'll see that `ionViewDidEnter` event fired only for the component D, and not for other three components.

Similar behavior can be seen, if you wrap TestPage in `withIonLifeCycle` - only the page will fire the event, and not the components inside it.

**Related code:**

**Other information:**

**Ionic info:**

```
Ionic:

   Ionic CLI       : 5.4.2
   Ionic Framework : @ionic/react 4.11.0

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v10.15.0 (C:\Program Files\nodejs\node.exe)
   npm    : 6.4.1
   OS     : Windows 10
```

**Comment by SatoAV** [comment]
I can assume, that this HOC was probably not designed to be used many times in one view, but I can't see other ways to have independent components that react to user entering the page.

**Comment by SatoAV** [comment]
@elylucas Hello!
Any news on this issue? Can we expect a fix anytime soon?

**Comment by elylucas** [comment]
@SatoAV taking a look at it today

**Comment by elylucas** [comment]
@SatoAV got a fix coming in 4.11.4 (being released today or tomorrow) that will allow multiple methods to subscribe to the lifecycle events.

Note, though, that the events fire for the main IonPage component (event the sub components). So if you subscribe to say IonViewWillEnter in a sub component inside of an IonPage, the event will fire when the IonPage comes into view and not the sub component.

**Comment by SatoAV** [comment]
> Note, though, that the events fire for the main IonPage component (event the sub components). So if you subscribe to say IonViewWillEnter in a sub component inside of an IonPage, the event will fire when the IonPage comes into view and not the sub component.

@elylucas That is exactly what I need it to do, so it's totally OK :)
Thanks for the update!

**Comment by elylucas** [comment]
fix released in 4.11.4
